src/backend_model.cc

```diff
@@ -485,17 +519,26 @@ TritonModel::PrepareInstances(
           }
         }
 
-        // The requested instance did not match an existing instance. Create a
-        // new instance.
-        std::shared_ptr<TritonModelInstance> new_instance;
-        LOG_VERBOSE(2) << "Creating model instance named '" << instance_name
-                       << "' with device id '" << is.device_id_ << "'";
-        RETURN_IF_ERROR(TritonModelInstance::CreateInstance(
-            this, instance_name, signature, is.kind_, is.device_id_,
-            profile_names, passive, is.policy_name_, *is.rate_limiter_config_,
-            secondary_devices, &new_instance));
-        added_instances->push_back(new_instance);
-        RegisterBackgroundInstance(std::move(new_instance), passive);
+        // Note that the local variables should be captured by value
+        creation_results.emplace_back(
+            std::async(launch_policy, [=, &instance_mu]() {
```

Comment:
Is it okay to copy all local variables? `creation_results` will be copied?

Reply:
According to docs/examples I read:

> A default capture (also called a capture-default) captures all variables that are mentioned in the lambda. Variables not mentioned in the lambda are not captured if a default capture is used.

A couples refs:
- https://learn.microsoft.com/en-us/cpp/cpp/lambda-expressions-in-cpp?view=msvc-170#:~:text=You%20can%20use%20a%20capture%2Ddefault%20mode%20to%20indicate%20how%20to%20capture%20any%20outside%20variables%20referenced%20in%20the%20lambda%20body
- https://www.learncpp.com/cpp-tutorial/lambda-captures/#:~:text=A%20default%20capture%20(also%20called,use%20a%20capture%20value%20of%20%3D%20.

I'm happy to make it more explicit if you'd prefer, I'm still cleaning up and finalizing on some finer details like this.

Reply:
@GuanLuo does 👍 you want more explicit? Or you're happy with it?

Reply:
How about using `&` instead of `=`?